clients/src/test/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImplTest.java

```diff
@@ -683,7 +681,10 @@ public void testDelayedMetadataUsedToCompleteAssignment() {
         receiveAssignment(newAssignment, membershipManager);
         membershipManager.poll(time.milliseconds());
 
-        verifyReconciliationNotTriggered(membershipManager);
+        // We bumped the local epoch, so new reconciliation is triggered
```

Comment:
This new reconciliation triggered is an example of what I was referring to in the [comment](https://github.com/apache/kafka/pull/15511/files#r1523858796) above, that seems to complicate the flow with 2 rebalances instead of 1. I wouldn't expect a member rebalancing/reconciling at this point (nothing changed for him yet, t2 not in metadata), and then reconciling/rebalance again once it discovers it.